contrib/fetch-offline-data/instana/fetch_instana_app_metrics.py

```diff
@@ -0,0 +1,140 @@
+
+    response = requests.get(instana_api_url, headers=headers, verify=False)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        logging.error("Error fetching plugins:", response.text)
+        return None
+
+def fetch_instana_application_metrics(url, token, metric_ids, start_time, end_time, granularity=1):
+    instana_api_url = f"{url}/api/application-monitoring/metrics/applications"
+
+    s_time = int(start_time.timestamp()) * 1000
+    f_time = int(end_time.timestamp()) * 1000
+    result = []
+    while s_time < f_time:
+        e_time = s_time + granularity * 600 * 1000
+        logging.info(f"Getting metric from {s_time} to {e_time}")
+        params = {
+            "start": s_time,  # Convert to milliseconds
+            "end": e_time,  # Convert to milliseconds
+        }
+
+        headers = {
+            "Authorization": f"apiToken {token}",
+            "Content-Type": "application/json"
+        }
+        metrics = [{"metric" : i, "aggregation":"PER_SECOND", "granularity": granularity} for i in metric_ids]
+        body = {
+            "metrics": metrics,
+            "order":{"by": "timestamp", "direction": "DESC"},
+            "timeFrame": {"to": params["end"], "windowSize": params["end"] - params["start"],
+            "pagination": {"page":1, "pageSize" : 200}},
+        }
+        parameters = {'offline': True}
+        response = requests.post(instana_api_url, params=parameters, headers=headers, json=body, verify=False)
+        if response.status_code == 200:
+            result.append(response.json()["items"])
+            total_hits = response.json()['totalHits']
+            if total_hits > 1:
+                logging.info(f"\tFound data in {total_hits} pages")
+                for page in range(2, total_hits):
+                    body['pagination'] = {"page":page, "pageSize" : 200}
+                    response = requests.post(instana_api_url, params=parameters, headers=headers, json=body, verify=False)
+                    if response.status_code == 200:
+                        result.append(response.json()["items"])
+                    else:
+                        logging.error("Error fetching metrics:", response.text)
+                        continue
+        else:
+            logging.error("Error fetching metrics:", response.text)
+            continue
+        s_time = e_time
+
+    return result
+
+def fetch_instana_metrics(url, token, start_time, end_time, granularity=1, folder = "data"):
+    metrics_catalog = fetch_instana_application_catalog(url, token[0])
+    if metrics_catalog is None:
+        logging.error(f"No application metrics available")
+        return 
+
+    metric_ids = list(map(lambda item: item['metricId'], metrics_catalog))
+
+    metrics_api_limit = 5
+    counter = 0
+
+    for i in range(0, len(metric_ids), metrics_api_limit):
+        token_ins = token[counter % len(token)]
+        metric_ids_subset = metric_ids[i:i + metrics_api_limit]
```

Comment:
If `len(metric_ids)` is not divisible by `metrics_api_limit`, it looks like this might go out of bounds when `i` reaches `len(metric_ids)`.